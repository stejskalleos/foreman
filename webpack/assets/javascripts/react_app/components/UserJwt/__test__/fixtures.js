import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
