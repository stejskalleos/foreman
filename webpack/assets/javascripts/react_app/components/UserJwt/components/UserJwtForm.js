import React, { useState } from 'react';
import { Grid, Dropdown, MenuItem, Button } from 'patternfly-react';
import PropTypes from 'prop-types';

import { translate as __ } from '../../../common/I18n';
import {
  UNIT_NEVER,
  UNIT_HOURS,
  UNIT_DAYS,
  UNIT_WEEKS,
  UNIT_MONTHS,
} from '../UserJwtConstants';

const UserJwtForm = ({ handleSubmit }) => {
  const expirationUnits = [
    { label: __('Never'), value: UNIT_NEVER },
    { label: __('Hours'), value: UNIT_HOURS },
    { label: __('Days'), value: UNIT_DAYS },
    { label: __('Weeks'), value: UNIT_WEEKS },
    { label: __('Months'), value: UNIT_MONTHS },
  ];

  const [expirationUnit, setExpirationUnit] = useState(expirationUnits[0]);
  const [expirationValue, setExpirationValue] = useState('');

  const changeUnit = u => {
    if (u.value === UNIT_NEVER) {
      setExpirationValue('');
    }
    setExpirationUnit(u);
  };

  return (
    <React.Fragment>
      <Grid.Row>
        <Grid.Col md={8}>
          <label className="control-label col-md-2">{__('Expiration')}</label>
          <div className="user-jwt-form">
            <div className="input-group user-jwt-expiration">
              <input
                type="number"
                className="jwt-expiration-value form-control"
                min={1}
                disabled={expirationUnit.value === UNIT_NEVER}
                value={expirationValue}
                placeholder={__('Expire in')}
                onChange={e => setExpirationValue(e.target.value)}
              />
              <div className="input-group-btn">
                <Dropdown id="expiration_unit">
                  <Dropdown.Toggle>{expirationUnit.label}</Dropdown.Toggle>
                  <Dropdown.Menu id="settings-dropdown">
                    {expirationUnits.map(unit => (
                      <MenuItem
                        key={unit.value}
                        onClick={() => changeUnit(unit)}
                      >
                        {unit.label}
                      </MenuItem>
                    ))}
                  </Dropdown.Menu>
                </Dropdown>
                <Button
                  bsStyle="success"
                  onClick={() => handleSubmit(expirationUnit, expirationValue)}
                >
                  {__('Generate')}
                </Button>
              </div>
            </div>
          </div>
        </Grid.Col>
      </Grid.Row>
    </React.Fragment>
  );
};

UserJwtForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default UserJwtForm;
