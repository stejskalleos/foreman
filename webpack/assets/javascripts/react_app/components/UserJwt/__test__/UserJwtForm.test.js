import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';
import UserJwtForm from '../components/UserJwtForm';
import { formProps } from './fixtures';

describe('UserJwtForm', () => {
  testComponentSnapshotsWithFixtures(UserJwtForm, { 'render ok': formProps });
});
