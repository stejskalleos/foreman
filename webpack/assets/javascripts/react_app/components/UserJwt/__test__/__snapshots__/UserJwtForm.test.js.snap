// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserJwtForm render ok 1`] = `
<Fragment>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={8}
    >
      <label
        className="control-label col-md-2"
      >
        Expiration
      </label>
      <div
        className="user-jwt-form"
      >
        <div
          className="input-group user-jwt-expiration"
        >
          <input
            className="jwt-expiration-value form-control"
            disabled={true}
            min={1}
            onChange={[Function]}
            placeholder="Expire in"
            type="number"
            value=""
          />
          <div
            className="input-group-btn"
          >
            <ForwardRef
              id="expiration_unit"
            >
              <DropdownToggle
                bsClass="dropdown-toggle"
                bsRole="toggle"
                open={false}
                useAnchor={false}
              >
                Never
              </DropdownToggle>
              <DropdownMenu
                bsClass="dropdown-menu"
                bsRole="menu"
                id="settings-dropdown"
                pullRight={false}
              >
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  header={false}
                  key="never"
                  onClick={[Function]}
                >
                  Never
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  header={false}
                  key="hours"
                  onClick={[Function]}
                >
                  Hours
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  header={false}
                  key="days"
                  onClick={[Function]}
                >
                  Days
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  header={false}
                  key="weeks"
                  onClick={[Function]}
                >
                  Weeks
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  header={false}
                  key="months"
                  onClick={[Function]}
                >
                  Months
                </MenuItem>
              </DropdownMenu>
            </ForwardRef>
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="success"
              disabled={false}
              onClick={[Function]}
            >
              Generate
            </Button>
          </div>
        </div>
      </div>
    </Col>
  </Row>
</Fragment>
`;
